corehq/apps/linked_domain/remote_accessors.py

```diff
@@ -57,7 +61,9 @@ def _convert_app_from_remote_linking_source(app_json):
     return app
 
 
-def pull_missing_multimedia_for_app(app):
+@task(queue='background_queue')
```

Comment:
This can be called by task and also directly.
I had quite some back and forth on this. 
One thing that bugs me is that we still need to load the app again even if it's called directly.
One way i was thinking was to have a method that is called and has args like `notify_user`/`email` and `async` and that decides what to do next to directly pull mm or just enqueue a task, that way things would be simpler. This is similar to what we discussed in `make_async_build` stuff.

Reply:
Might not be relevant any more, but a possible approach to this problem is to have the `task` function accept an app_id, then fetch the app and pass it to the _actual_ function.  That way you can use the actual function directly without re-fetching the app, but the task will only need the app_id.

Reply:
Ya so that needs me to add another function specifically for task.  I think thats what i eventually did or this whole situation got ruled out.